Emby.Server.Implementations/Networking/NetworkManager.cs

```diff
@@ -636,6 +636,65 @@ public bool TryParseIpAddress(string ipAddress, out IpAddressInfo ipAddressInfo)
             return false;
         }
 
+        public bool IsInSameSubnet(IpAddressInfo address1, IpAddressInfo address2, IpAddressInfo subnetMask)
+        {
+             IPAddress network1 = GetNetworkAddress(ToIPAddress(address1), ToIPAddress(subnetMask));
+             IPAddress network2 = GetNetworkAddress(ToIPAddress(address2), ToIPAddress(subnetMask));
+             return network1.Equals(network2);
+        }
+
+        private IPAddress GetNetworkAddress(IPAddress address, IPAddress subnetMask)
+        {
+            byte[] ipAdressBytes = address.GetAddressBytes();
+            byte[] subnetMaskBytes = subnetMask.GetAddressBytes();
+
+            if (ipAdressBytes.Length != subnetMaskBytes.Length)
+                throw new ArgumentException("Lengths of IP address and subnet mask do not match.");
```

Comment:
```suggestion
            {
                throw new ArgumentException("Lengths of IP address and subnet mask do not match.");
            }
```
per our convention on using brackets I think